packages/engine/src/lib/services/variable-service.ts

```diff
@@ -105,4 +106,28 @@ export class VariableService {
       censorConnections
     );
   }
+
+  validateAndCast(resolvedInput: any, props: PiecePropertyMap): { result: any, errors: Record<string, any> } {
+    const errors: Record<string, string | Record<string, string>> = {};
+    const clonedInput = JSON.parse(JSON.stringify(resolvedInput));
+    for (const [key, value] of Object.entries(resolvedInput)) {
+      const property = props[key];
+      if (property.type === PropertyType.NUMBER && value) {
```

Comment:
This will not parse zeros '0'

Reply:
check if value is not equal to undefined or null or empty-string

Reply:
Added tests